xbmc/settings/AdvancedSettings.cpp

```diff
@@ -232,8 +232,8 @@ void CAdvancedSettings::Initialize()
   //m_videoStackRegExps.push_back("(.*?)([ ._-]*[0-9])(.*?)(\\.[^.]+)$");
 
   m_tvshowEnumRegExps.clear();
-  // foo.s01.e01, foo.s01_e01, S01E02 foo, S01 - E02
-  m_tvshowEnumRegExps.push_back(TVShowRegexp(false,"s([0-9]+)[ ._-]*e([0-9]+(?:(?:[a-i]|\\.[1-9])(?![0-9]))?)([^\\\\/]*)$"));
+  // foo.s01.e01, foo.s01_e01, S01E02 foo, S01 - E02, S01xE02
+  m_tvshowEnumRegExps.push_back(TVShowRegexp(false,"s([0-9]+)[ ._-x]*e([0-9]+(?:(?:[a-i]|\\.[1-9])(?![0-9]))?)([^\\\\/]*)$"));
```

Comment:
Yep, you're correct on both counts. The new expression will most likely accept `w` (or anything else between `_` and `x`, using `echo "s03we04" | grep -E "s([0-9]+)[ ._-x]*e([0-9]+)"` as a simple but indicative test), and positioning `-` at the end (or beginning) eliminates that confusion.